pandas/tseries/tools.py

```diff
@@ -410,21 +434,82 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                 raise e
 
     if arg is None:
-        return arg
-    elif isinstance(arg, tslib.Timestamp):
-        return arg
+        return None
+
+    # handle origin
+    if origin == 'julian':
+
+        original = arg
+        j0 = tslib.Timestamp(0).to_julian_date()
+        if unit != 'D':
+            raise ValueError("unit must be 'D' for origin='julian'")
+        try:
+            arg = arg - j0
+        except:
+            raise ValueError("incompatible 'arg' type for given "
+                             "'origin'='julian'")
+
+        # premptively check this for a nice range
+        j_max = tslib.Timestamp.max.to_julian_date() - j0
+        j_min = tslib.Timestamp.min.to_julian_date() - j0
+        if np.any(arg > j_max) or np.any(arg < j_min):
+            raise tslib.OutOfBoundsDatetime(
+                "{original} is Out of Bounds for "
+                "origin='julian'".format(original=original))
```

Comment:
yes, but that is handled below (this is *only* for the julian section). I do raise the same error (w/o the julian reference)